### PR TITLE
handle tweet urls by link_resolver

### LINKED ISSUE #24: Thumbnails for Twitter links
image

Currently Twitter  links have no extra information. It would be nice to get a preview of the text and potentially the images linked in the tweet.

Here is an example tweet with an image: https://twitter.com/nymnion/status/1273584566826086400
And a plain tweet without anything fancy: https://twitter.com/nymnion/status/1269222603803234304

### PR DESCRIPTION
**Pull request by the-vesuvius**
partially closes #24 
doesn't get a thumbnail but fetches tweet's text and author's username
I didn't add any new dependencies (its just one HTTP request)
introduced new env var CHATTERINO_TWITTER_BEARER_KEY.

### CODE REVIEW

File: link_resolver_twitter.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+}
+
+func init() {
+    bearerKey, exists := os.LookupEnv("CHATTERINO_TWITTER_BEARER_KEY")
+    if !exists {
+       log.Println("No CHATTERINO_TWITTER_BEARER_KEY specified, won't do special responses for twitter")
+       return
+    }
+
+    tooltipTemplate, err := template.New("youtubeTooltip").Parse(tweeterTooltip)
```
Comment (pajlada) [inline link_resolver_twitter.go:68]: should be "tweetTooltip", not "youtubeTooltip"

File: link_resolver_twitter.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+}
+
+type tweetTooltipData struct {
+    Text      string
+    Username  string
+    Thumbnail string
+}
+
+func init() {
+    bearerKey, exists := os.LookupEnv("CHATTERINO_TWITTER_BEARER_KEY")
```
Comment (pajlada) [inline link_resolver_twitter.go:62]: Rename the environment variable to `CHATTERINO_API_TWITTER_BEARER_TOKEN`

Comment (the-vesuvius) [reply link_resolver_twitter.go]: fixed

File: link_resolver_twitter.go (hunk 1 above)
Comment (the-vesuvius) [reply link_resolver_twitter.go]: fixed

**Comment by pajlada** [comment]
@the-vesuvius How do you generate a bearer token? I tried following the instructions from here: https://developer.twitter.com/en/docs/basics/authentication/oauth-2-0/bearer-tokens but it doesn't work for me

**Comment by the-vesuvius** [comment]
@pajlada actually I didn't generate anything myself
After my twitter app got approved I went to the developer portal and the keys were presented to me

**Comment by the-vesuvius** [comment]
Now I'm able revoke/regenerate it via developer portal

**Comment by pajlada** [comment]
@the-vesuvius What key from the app are you specifically using? A full guide for how to use this would be great

**Comment by the-vesuvius** [comment]
@pajlada I'm using the Bearer Token
https://paste.pics/9LALH

>>A full guide for how to use this would be great
not sure what to provide here, its just copy the value and set it as CHATTERINO_API_TWITTER_BEARER_TOKEN env var

The problem here might be that twitter changed how the apps work (it looks like, I'm not sure how everything used to be before). So maybe if you are using an older app you can't see the Bearer Token. Are you using a new twitter app or an older one?

For testing purposes I can share my key with you

**Comment by pajlada** [comment]
Turns out that API is an experimental endpoint that I and others might not have access to - I'd prefer if this uses the "standard" API endpoint https://developer.twitter.com/en/docs/tweets/post-and-engage/api-reference/get-statuses-show-id

The credentials we want then is probably the two values used to generate a Bearer Token (so the consumer key and secret)

**Comment by pajlada** [comment]
Closing in favour of #38 
Thank you for your contribution regardless!

**Pull request closed without merging**